### ISSUE TITLE
CSW Request fails with NoneType error, missing section from service causes fault

### ISSUE THREAD

**Issue by wsidl** [geopython/OWSLib#243]
A CSW request will receive a result with no "ServiceProvider" will cause the application to fault:
```
  File "cachedb.py", line 16, in __init__
    self.csw_conn = owslib.csw.CatalogueServiceWeb(source_address)
  File "C:\Python34\lib\site-packages\owslib-0.8.dev0-py3.4.egg\owslib\csw.py", line 95, in __init__
    self.provider=ows.ServiceProvider(val,self.owscommon.namespace)
  File "C:\Python34\lib\site-packages\owslib-0.8.dev0-py3.4.egg\owslib\ows.py", line 80, in __init__
    val = self._root.find(util.nspath('ProviderName', namespace))
```

In line 94, it tries to eval "ows:ServiceProvider" and returns an empty object. This will cause the application to fault. Adding a catch for each scenario should at least resolve the issue:
```
val = self._exml.find(util.nspath_eval('ows:ServiceProvider', namespaces))
if not val is None:
    self.provider=ows.ServiceProvider(val, self.owscommon.namespace)
```

**Comment by tomkralidis** [comment]
Thanks for the report.  Can you provide a testcase which can reproduce this issue?

**Comment by wsidl** [comment]
Looking at the code again, this may be resolved, but may be that I made another changed and lost it.

I'm connecting to a CARIS Spatial Fusion Enterprise Server. There is a public access server that you should be able to test with:
http://demo.caris.com/sfs/services/ows/csw

This bit of code will test this connection:
```
import owslib.csw
conn = owslib.csw.CatalogueServiceWeb('http://demo.caris.com/sfs/services/ows/csw')
print(conn.identification.type)
```

Before, I would have gotten the fault noted above. Should print out "CSW".
BTW, will also need to change the "Accept" header values in order to complete the connection as well.

**Comment by tomkralidis** [comment]
@wsidl fix pushed to e4f199fd9f348c23da2c632c864943013e6ab9e1.  Can you verify?

**Comment by wsidl** [comment]
Works! Thanks
